libs/galley-types/src/Galley/Types/Teams.hs

```diff
@@ -258,9 +263,16 @@ newTeam tid uid nme ico bnd = Team tid uid nme ico Nothing bnd
 newTeamList :: [Team] -> Bool -> TeamList
 newTeamList = TeamList
 
-newTeamMember :: UserId -> Permissions -> TeamMember
+newTeamMember :: UserId -> Permissions -> Maybe (UserId, UTCTimeMillis) -> TeamMember
 newTeamMember = TeamMember
 
+newTeamMemberRaw :: MonadThrow m => UserId -> Permissions -> Maybe UserId -> Maybe UTCTimeMillis -> m TeamMember
+newTeamMemberRaw uid perms (Just invu) (Just invt) = pure $ TeamMember uid perms (Just (invu, invt))
+newTeamMemberRaw uid perms Nothing Nothing         = pure $ TeamMember uid perms Nothing
+newTeamMemberRaw _ _ _ _ = throwM $ ErrorCall "TeamMember with incomplete metadata."
+    -- TODO: question for the reviewer: is this exception type what we want to use here, and on the
```

Comment:
What will the HTTP response look like in such cases?

Reply:
i've added two more, hopefully non-controversial commits and will merge when CI is green.